contracts/FeeDisbursal/FeeDisbursal.sol

```diff
@@ -288,100 +372,111 @@ contract FeeDisbursal is
     /// @param token address of the claimable token
-    /// @param spender address of the spender that will receive the funds
+    /// @param claimer address of the claimer that will receive the funds
     /// @param until until what index to claim to, use UINT256_MAX for all
     /// @param onlyRecycleable show only claimable funds that can be recycled
-    function claimablePrivate(address token, address spender, uint256 until, bool onlyRecycleable)
-    internal
-    constant
-    returns (uint256 claimableAmount, uint256 lastIndex)
+    /// @return a tuple of (amount claimed, total amount disbursed, next disbursal index to be claimed)
+    function claimablePrivate(address token, ITokenSnapshots proRataToken, address claimer, uint256 until, bool onlyRecycleable)
+        internal
+        constant
+        returns (uint256 claimableAmount, uint256 totalAmount, uint256 nextIndex)
     {
-        lastIndex = min(until, _disbursals[token].length);
-        claimableAmount = 0;
-        uint256 currentIndex = 0;
-        for (currentIndex = _disbursalProgress[token][spender]; currentIndex < lastIndex; currentIndex += 1) {
-            Disbursal storage disbursal = _disbursals[token][currentIndex];
-            // in case of just determining the recyclable amount of tokens, break when we
-            // cross this time
-            if ( onlyRecycleable && disbursal.recycleableAfterTimestamp > block.timestamp )
-                break;
-            BasicSnapshotToken proRataToken = BasicSnapshotToken(disbursal.proRataToken);
+        nextIndex = min(until, _disbursals[token][proRataToken].length);
+        uint256 currentIndex = _disbursalProgress[token][proRataToken][claimer];
+        uint256 currentSnapshotId = proRataToken.currentSnapshotId();
+        for (; currentIndex < nextIndex; currentIndex += 1) {
+            Disbursal storage disbursal = _disbursals[token][proRataToken][currentIndex];
             uint256 snapshotId = disbursal.snapshotId;
             // do not pay out claims from the current snapshot
-            if ( snapshotId == proRataToken.currentSnapshotId() )
+            if ( snapshotId == currentSnapshotId )
                 break;
-            uint256 proRataTokenTotalSupply = proRataToken.totalSupplyAt(snapshotId);
-            uint256 proRataSpenderBalance = proRataToken.balanceOfAt(spender, snapshotId);
-            if (proRataTokenTotalSupply == 0 || proRataSpenderBalance == 0) continue;
-            // this should round down, so we should not be spending more than we have in our balance
-            claimableAmount += proportion(disbursal.amount, proRataSpenderBalance, proRataTokenTotalSupply);
+            // in case of just determining the recyclable amount of tokens, break when we
+            // cross this time, this also assumes disbursal.recycleableAfterTimestamp in each disbursal is the same or increases
+            // in case it decreases, recycle will not be possible until 'blocking' disbursal also expires
```

Comment:
what exactly does this comment mean?

Reply:
if you do disbursal with short time to recycle then with long time and then another with short time, you will not be able to recycle 1 and 3 when time passes because disbursal 2 will not yet expire and you will not reach 3. this is how disbursalProgress works. I'll write test for it